fastapi_utils/crud_base.py

```diff
@@ -0,0 +1,160 @@
+from decimal import Decimal
+from enum import Enum
+from typing import Dict, Generic, List, Optional, Type, TypeVar, Union
+
+from fastapi.encoders import jsonable_encoder
+from pydantic import BaseModel
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import Session
+from sqlalchemy_filters import apply_filters, apply_sort
+
+from fastapi_utils.camelcase import snake2camel
+
+Base = declarative_base()
```

Comment:
As noted above, for multiple reasons I would much prefer if it were possible for the user to provide their own `declarative_base()`, rather than using the one here.

I think it may be challenging to get the type hinting right, but I think the functionality benefits are worth the associated challenge with the `ModelType` TypeVar.

I suspect in the worst case there's probably something we can do with an `if TYPE_CHECKING:` block to make it work nicely for most/all cases.